unified_planning/model/ma_problem.py

```diff
@@ -0,0 +1,250 @@
+# Copyright 2021 AIPlan4EU project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+'''This module defines the MultiAgentProblem class.'''
+
+import unified_planning
+import sys
+import copy
+from unified_planning.shortcuts import *
+import unified_planning.model.operators as op
+from unified_planning.model.types import domain_size, domain_item
+from unified_planning.environment import get_env, Environment
+from unified_planning.walkers import OperatorsExtractor
+from fractions import Fraction
+from typing import List, Dict, Set, Union, Optional
+from unified_planning.model.problem import Problem
+from unified_planning.exceptions import UPProblemDefinitionError, UPTypeError, UPValueError, UPExpressionDefinitionError
+from unified_planning.model.agent import Agent
+from unified_planning.io.pddl_writer import PDDLWriter
+from unified_planning.environment import Environment
+from unified_planning.model.environment_ma import Environment_
+import collections
+from typing import List, Union
+import re
+
+
+class MultiAgentProblem(Problem):
+    '''Represents a planning MultiAgentProblem.'''
+    def __init__(self, *args, **kwargs):
+        super(MultiAgentProblem, self).__init__(*args, **kwargs)
+
+    env_ = None
```

Comment:
These are class fields (static fields in the java lingo)! You need to put these inside the `__init__` as `self.env_`, `self.agents_list`, ... etc to have different values in different instances, otherwise you can only have ONE MultiAgentProblem at a time!